### PR TITLE
Add `elem` combinator to API

### PR DESCRIPTION
**Pull request by emilypi**
This PR adds the `elem` combinator to the API.

### CODE REVIEW

File: Data/Text.hs [hunk 1]
```diff
@@ -1488,6 +1490,13 @@ chunksOf k = go
 -------------------------------------------------------------------------------
 -- ** Searching with a predicate
 
+-- | /O(n)/ The 'elem' function takes a character and a 'Text', and
+-- returns 'True' if the element is found in the given 'Text', or
+-- 'False' otherwise.
+elem :: Char -> Text -> Bool
+elem c t = isJust (find (== c) t)
```
Comment (hvr) [inline Data/Text.hs:1497]: Why not define this in terms of the `S.any` primitive that `Data.Text.any` is based upon?

```
any :: (Char -> Bool) -> Text -> Bool
any p t = S.any p (stream t)
{-# INLINE any #-}
```

?

**Comment by hvr** [comment]
So I'm slightly on the fence about this; for one, `elem` is trivially definable in terms of `any`

```hs
elem c = any (==c)
```

and I'm not sure if this meets the https://wiki.haskell.org/Fairbairn_threshold standard.

Fwiw, there's appears to be lack of evidence that this is *needed* insofar as nobody seems to have missed having this trivial definition in `Data.Text` for over a decade...

However, there's precedent in e.g. `Data.ByteString.elem` and `Data.Vector.elem` which do provide this standard verb; so for the sake of API symmetry/uniformity among vector/list-ish data-structures, it'd make sense to add this (redundant) verb, as there might be some level of expectation of it existing.

File: Data/Text.hs (hunk 1 above)
Comment (emilypi) [reply Data/Text.hs:1497]: Lack of knowledge of the `text` library is the reason. If that's the preferred way, let's do it.

**Comment by emilypi** [comment]
> So I'm slightly on the fence about this; for one, elem is trivially definable in terms of any

My reasoning is precisely that this is the API that's defined for every other "stringlike" object. It passes the Fairbarn test on account of it's expected to be there, eminently useful (perhaps less so for text), and it homogenizes the APIs across libraries. You may object to it on the bounds that  it's a small addition, but it was *surprising* that it was not already.

File: Data/Text.hs (hunk 1 above)
Comment (hvr) [reply Data/Text.hs:1497]: @emilypi let's do that then :-)

**Comment by emilypi** [comment]
@hvr updated and it should be ready to go

**Comment by emilypi** [comment]
@phadej too, if you're more active

**Comment by emilypi** [comment]
@hvr rebased to master so we should be able to just merge with no problems

**Comment by Lysxia** [comment]
This looks good to me (conclusion: we want this for uniformity with other collection-like APIs). Ping @haskell/text for another approval to merge this.

**Review by Bodigrim — APPROVED** [review]
Adding a test would be nice.

**Pull request merged**